### PR TITLE
crypto/kzg4844: use the new trusted setup file and format

### PR DESCRIPTION
**Pull request by kevaundray**
**Changes made:**

_Consensus specs_

As [this commit](https://github.com/ethereum/consensus-specs/commit/c5785899f7a8aba5e0872f8c444f58de56ef09c3) the trusted setup file was changed to use the official version that will be used on mainnet.

This trusted setup file also has a different format to the one that was previously being used.

_Cryptography libraries_

As of [this commit](https://github.com/crate-crypto/go-kzg-4844/commit/3621cc59f0c738db779a1672b20758d12d3ec302) and [this commit](https://github.com/ethereum/c-kzg-4844/commit/748283cced543c486145d5f3f38684becdfe3e1b) go-kzg and c-kzg are now conformant with the new file format and is running tests against the new consensus spec test vectors.

### CODE REVIEW

**Review by MariusVanDerWijden — APPROVED** [review]
LGTM, 
the sha256sum does not check out, because the `trusted_setup.json` has two additional spaces on each line compared to the original here: https://github.com/ethereum/consensus-specs/blob/c5785899f7a8aba5e0872f8c444f58de56ef09c3/presets/mainnet/trusted_setups/trusted_setup_4096.json
If you remove the superfluous spaces, the shasum checks out.
I would prefer if you committed the original version, just so its easier to verify that this is the correct ceremony

**Comment by holiman** [comment]
Please submit a `trusted_setup.json` which matches the expected signature

File: go.sum [hunk 1]
```diff
@@ -185,6 +187,8 @@ github.com/envoyproxy/protoc-gen-validate v0.1.0/go.mod h1:iSmxcyjqTsJpI2R4NaDN7
 github.com/etcd-io/bbolt v1.3.3/go.mod h1:ZF2nL25h33cCyBtcyWeZ2/I3HQOfTP+0PIEvHjkjCrw=
 github.com/ethereum/c-kzg-4844 v0.3.1 h1:sR65+68+WdnMKxseNWxSJuAv2tsUrihTpVBTfM/U5Zg=
 github.com/ethereum/c-kzg-4844 v0.3.1/go.mod h1:VewdlzQmpT5QSrVhbBuGoCdFJkpaJlO1aQputP83wc0=
+github.com/ethereum/c-kzg-4844 v0.3.2-0.20231019020040-748283cced54 h1:jDyQvQjauRyb7TJAF9W7J3NOjn3ukXahd3l+rd1Fak8=
+github.com/ethereum/c-kzg-4844 v0.3.2-0.20231019020040-748283cced54/go.mod h1:VewdlzQmpT5QSrVhbBuGoCdFJkpaJlO1aQputP83wc0=
```
Comment (jtraglia) [inline go.sum:191]: This can be v0.4.0 now. Also, shouldn't the old version (v0.3.1) be removed? Same for go-kzg v0.3.0.

**Review by lightclient — APPROVED** [review]
LGTM

for those who wanna verify at home:

```console
$ wget https://raw.githubusercontent.com/ethereum/go-ethereum/4e16f39e151963f225e70d55712c9ec337effbbf/crypto/kzg4844/trusted_setup.json
$ shasum -a 256 trusted_setup.json
0229b43f4fac9b17374809520eb621b5ee1a7f74547e7d36918e7d4b122e178d
```

Compare with the official PR: https://github.com/ethereum/consensus-specs/pull/3521

**Comment by MariusVanDerWijden** [comment]
CI is red, because the execution-spec tests still use the old format

**Comment by MariusVanDerWijden** [comment]
```diff
diff --git a/build/checksums.txt b/build/checksums.txt
index c752fb8ccd..dd8a9cdbf0 100644
--- a/build/checksums.txt
+++ b/build/checksums.txt
@@ -1,9 +1,9 @@
 # This file contains sha256 checksums of optional build dependencies.
 
-# version:spec-tests 1.0.5
+# version:spec-tests 1.0.6
 # https://github.com/ethereum/execution-spec-tests/releases
-# https://github.com/ethereum/execution-spec-tests/releases/download/v1.0.5/
-d4fd06a0e5f94beb970f3c68374b38ef9de82d4be77517d326bcf739c3cbf3a2  fixtures_develop.tar.gz
+# https://github.com/ethereum/execution-spec-tests/releases/download/v1.0.6/
+485af7b66cf41eb3a8c1bd46632913b8eb95995df867cf665617bbc9b4beedd1  fixtures_develop.tar.gz
 
 # version:golang 1.21.3
 # https://go.dev/dl/
```
This will update the execution-spec-tests to the newest release v1.0.6 which adds the correct 4844 tests (see https://github.com/ethereum/execution-spec-tests/releases/tag/v1.0.6)

There is also a static test case failing (core/vm/testdata/precompiles/pointEvaluation.json) I'll try to find a fix for that

**Comment by MariusVanDerWijden** [comment]
```diff
diff --git a/core/vm/testdata/precompiles/pointEvaluation.json b/core/vm/testdata/precompiles/pointEvaluation.json
index 93fc66d836..f2a4fd7a72 100644
--- a/core/vm/testdata/precompiles/pointEvaluation.json
+++ b/core/vm/testdata/precompiles/pointEvaluation.json
@@ -1,9 +1,10 @@
 [
   {
-    "Input": "01d18459b334ffe8e2226eef1db874fda6db2bdd9357268b39220af2d59464fb564c0a11a0f704f4fc3e8acfe0f8245f0ad1347b378fbf96e206da11a5d3630624d25032e67a7e6a4910df58
34b8fe70e6bcfeeac0352434196bdf4b2485d5a1978a0d595c823c05947b1156175e72634a377808384256e9921ebf72181890be2d6b58d4a73a880541d1656875654806942307f266e636553e94006d11423f2
688945ff3bdf515859eba1005c1a7708d620a94d91a1c0c285f9584e75ec2f82a",
+    "Input": "01e798154708fe7789429634053cbf9f99b619f9f084048927333fce637f549b564c0a11a0f704f4fc3e8acfe0f8245f0ad1347b378fbf96e206da11a5d3630624d25032e67a7e6a4910df58
34b8fe70e6bcfeeac0352434196bdf4b2485d5a18f59a8d2a1a625a17f3fea0fe5eb8c896db3764f3185481bc22f91b4aaffcca25f26936857bc3a7c2539ea8ec3a952b7873033e038326e87ed3e1276fd14025
3fa08e9fc25fb2d9a98527fc22a2c9612fbeafdad446cbc7bcdbdcd780af2c16a",
     "Expected": "000000000000000000000000000000000000000000000000000000000000100073eda753299d7d483339d80809a1d80553bda402fffe5bfeffffffff00000001",
     "Name": "pointEvaluation1",
     "Gas": 50000,
     "NoBenchmark": false
   }
 ]
 ```
 Heres the fix to the static test case

**Comment by MariusVanDerWijden** [comment]
Again I can't push it to your fork so you will need to apply the diff yourself @kevaundray

**Comment by kevaundray** [comment]
Strange you cannot edit -- I have the correct settings ticked

Screenshot 2023-10-21 at 13 46 34

**Review by holiman — APPROVED** [review]
LGTM, thanks @kevaundray !

**Pull request merged**